pan-domain-auth-verification/src/test/scala/com/gu/pandomainauth/PanDomainTest.scala

```diff
@@ -49,30 +60,48 @@ class PanDomainTest extends AnyFreeSpec with Matchers with Inside {
+      // The cookie expired 24 hours and 1 minute ago.
+      // Therefore, we are one minute after the end of the default 24-hour grace period.
+      val expiredAuthUser = authUser.copy(expires = now() minus PanDomain.DefaultApiGracePeriod.plusMinutes(1))
       val cookieData = CookieUtils.generateCookieData(expiredAuthUser, signingWith(testPrivateKey.key))
 
       authStatus(cookieData) shouldBe a [Expired]
     }
 
-    "returns `Expired` if the cookie has expired and is outside the grace period" in {
-      val expiredAuthUser = authUser.copy(expires = new Date().getTime - 86400)
+    "returns `Expired` if the cookie has expired and is outside a custom grace period" in {
+      val customGracePeriod = ofHours(2)
+      // The cookie expired 2 hours and 1 minute ago.
+      // Therefore, we are one minute after the end of the custom 2-hour grace period.
+      val expiredAuthUser = authUser.copy(expires = now() minus customGracePeriod.plusMinutes(1))
       val cookieData = CookieUtils.generateCookieData(expiredAuthUser, signingWith(testPrivateKey.key))
 
-      authStatus(cookieData) shouldBe a [Expired]
+      authStatus(cookieData, apiGracePeriod = Some(customGracePeriod)) shouldBe a [Expired]
     }
 
-    "returns `Expired` if cookie has not expired, but forceExpiry is set" in {
-      val validCookieData = CookieUtils.generateCookieData(authUser, signingWith(testPrivateKey.key))
-      authStatus(validCookieData, forceExpiry = true) shouldBe a [Expired]
+    "returns `GracePeriod` if the cookie has expired but is within the default grace period" in {
+      // The cookie expired 23 hour and 59 minutes ago.
+      // Therefore, we have one minute left of the default 24-hour grace period.
+      val expiredAuthUser = authUser.copy(expires = now() minus PanDomain.DefaultApiGracePeriod.minusMinutes(1))
+      val cookieData = CookieUtils.generateCookieData(expiredAuthUser, signingWith(testPrivateKey.key))
+
+      authStatus(cookieData) shouldBe a [GracePeriod]
     }
 
-    "returns `GracePeriod` if the cookie has expired but is within the grace period" in {
-      val expiredAuthUser = authUser.copy(expires = new Date().getTime - 3000)
+    "returns `GracePeriod` if the cookie has expired but is within a custom grace period" in {
+      val customGracePeriod = ofHours(2)
+      // The cookie expired 1 hour and 59 minutes ago.
+      // Therefore, we have one minute left of the custom 2-hour grace period.
+      val expiredAuthUser = authUser.copy(expires = now() minus customGracePeriod.minusMinutes(1))
       val cookieData = CookieUtils.generateCookieData(expiredAuthUser, signingWith(testPrivateKey.key))
 
-      authStatus(cookieData, apiGracePeriod = 3600) shouldBe a [GracePeriod]
+      authStatus(cookieData, apiGracePeriod = Some(customGracePeriod)) shouldBe a [GracePeriod]
+    }
+
+
+    "returns `Expired` if cookie has not expired, but forceExpiry is set" in {
```

Comment:
@rtyley this is the only mention of `forceExpiry` in the tests.

It makes it clear that when the cookie is fresh, setting `forceExpiry` to `true` will give an `AuthenticationStatus` of `Expired`.

However, it doesn't specify the behaviour wrt grace period.

It looks to me like the existing behaviour in `checkStatus` is that if you are in period C below (aka the grace period), then `forceExpiry` will never work!

```
Panda cookie:  issued       expires
               |-- 1 hour --|--24 hours------------------------------|
            A--|----B-------|--C-------------------------------------|--D----
```


This seems accidental. Why should the grace period "protect" you from forced expiry?

Reply:
One other useful thing I discovered: the backstory of `forceExpiry`, apparently just for debug purposes https://github.com/guardian/pan-domain-authentication/blob/a9e2027516ad1f7e424e7f2dd8b2c918c1482e76/pan-domain-auth-play/src/main/scala/com/gu/pandomainauth/action/Actions.scala#L105-L109

Reply:
> the backstory of `forceExpiry`, apparently just for debug purposes

Ah yeah, just found the PR:

* https://github.com/guardian/pan-domain-authentication/pull/117

...it's a convenient way to simulate an expired cookie, but I don't think it addresses the distinction between expired/grace-period/acceptable/fresh... what happens if you want to be in- or out - of the grace-period?

Possibly finishing off this debug tool might be an alternative - not as convenient to use though:

* https://github.com/guardian/pan-domain-authentication/pull/177

...but at least it would simplify the code in `checkStatus()`.

Reply:
So, I **think** this weirdness is fixed by the refactor in https://github.com/guardian/pan-domain-authentication/pull/188!

But we should add a test to that PR to check. Let's get this one merged first.